### PR TITLE
Rectify differences in the custom SSL CSR for LB and Smart Proxy for 3.8

### PR DESCRIPTION
**Pull request by AkshayGadhaveRH**
Same change as https://github.com/theforeman/foreman-documentation/pull/2832. Pushing separate PR for 3.8 as there was a merge conflict on this branch.

There are differences in the openssl.cnf files created for cert signing requests for LB and for Smart Proxy:

guides/common/modules/proc_creating-custom-ssl-certificates-for-smart-proxy-server.adoc > LB custom SSL CSR file guides/common/modules/proc_creating-a-custom-ssl-certificate.adoc > Smart Proxy custom SSL CSR file Trying to make these two files similar as far as possible.

This PR will also include changes to guides/common/modules/proc_creating-a-custom-ssl-certificate.adoc. However, there is another PR in progress for changes in this file: #2828

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2256021

* [X] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [ ] Foreman 3.10/Katello 4.12
* [ ] Foreman 3.9/Katello 4.11 (Satellite 6.15; orcharhino 6.8)
* [ ] Foreman 3.8/Katello 4.10
* [x] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [x] Foreman 3.6/Katello 4.8
* [x] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6/6.7)
* [ ] Foreman 3.4/Katello 4.6 (EL8 only)
* [ ] Foreman 3.3/Katello 4.5 on EL7 & EL8 (Satellite 6.12 on EL8 only; orcharhino 6.4/6.5 on EL8 only)
* We do not accept PRs for Foreman older than 3.3.

### CODE REVIEW

File: guides/common/assembly_configuring-capsule-custom-server-certificate.adoc [hunk 1]
```diff
@@ -13,8 +13,12 @@ To configure your {SmartProxyServer} with a custom certificate, complete the fol
 . xref:deploying-a-custom-ssl-certificate-to-{smart-proxy-context}-server_{smart-proxy-context}[]
 . xref:deploying-a-custom-ssl-certificate-to-hosts_{smart-proxy-context}[]
 
+:ssl-common-name: {smartproxy-example-com}
```
Comment (Lennonka) [inline guides/common/assembly_configuring-capsule-custom-server-certificate.adoc:16]:
```suggestion
:ProductName: {SmartProxyServer}
:ssl-common-name: {smartproxy-example-com}
```
Also all other places.

**Review by Lennonka — CHANGES_REQUESTED** [review]
You're missing the `ProductName` definition and it probably makes the build fail.

**Comment by AkshayGadhaveRH** [comment]
Hey @Lennonka,

Not sure what exactly is causing the `asciidoctor: WARNING: skipping reference to missing attribute: parent-context`

Strangely it complains about two files that I haven't touched at all:
> 
> asciidoctor: WARNING: common/modules/proc_configuring-smart-proxy-server-with-custom-ssl-certificates-to-generate-and-sign-puppet-certificates.adoc: line 2: section title out of sequence: expected level 4, got level 6
> asciidoctor: WARNING: common/modules/proc_configuring-remaining-smart-proxy-servers-with-custom-ssl-certificates.adoc: line 2: section title out of sequence: expected level 4, got level 6

File: guides/common/assembly_configuring-satellite-custom-server-certificate.adoc [hunk 2]
```diff
@@ -25,8 +25,13 @@ To configure your {ProjectServer} with a custom certificate, complete the follow
 . If you have external {SmartProxyServers} registered to {ProjectServer}, configure them with custom SSL certificates.
 For more information, see {InstallingSmartProxyDocURL}configuring-capsule-custom-server-certificate_{smart-proxy-context}[Configuring {SmartProxyServer} with a Custom SSL Certificate] in _{InstallingSmartProxyDocTitle}_.
 
+
```
Comment (maximiliankolb) [inline guides/common/assembly_configuring-satellite-custom-server-certificate.adoc:28]: Is this missing "ProductName" similar to https://github.com/theforeman/foreman-documentation/pull/3012/files#diff-b3c1e911c867b76feaa0e96e4320c9057a49ec4d0a776afe750faf65dd3305e2R16?

File: guides/common/modules/proc_creating-a-custom-ssl-certificate.adoc [hunk 3]
```diff
@@ -77,21 +105,37 @@ organizationalUnitName = _My_Organizational_Unit_Name_ <4>
 <3> Full name (example: New York)
 <4> Division responsible for the certificate (example: IT department)
 . Generate CSR:
```
Comment (maximiliankolb) [inline guides/common/modules/proc_creating-a-custom-ssl-certificate.adoc:107]:
```suggestion
. Generate CSR:
+
```

File: guides/common/modules/proc_creating-a-custom-ssl-certificate.adoc [hunk 4]
```diff
@@ -27,45 +32,68 @@ If you already have a private key for this {ProductName}, skip this step.
...
+[ v3_req ]
+basicConstraints = CA:FALSE
+keyUsage = digitalSignature, nonRepudiation, keyEncipherment, dataEncipherment
+extendedKeyUsage = serverAuth, clientAuth, codeSigning, emailProtection
+subjectAltName = @alt_names
+
+[alt_names] <2>
+DNS.1 = _{loadbalancer-example-com}_
+DNS.2 = _{smartproxy-example-com}_
+---
```
Comment (Lennonka) [inline guides/common/modules/proc_creating-a-custom-ssl-certificate.adoc:84]:
```suggestion
----
```
This should fix it :)

File: guides/doc-Configuring_Load_Balancer/master.adoc [hunk 5]
```diff
@@ -32,16 +33,28 @@ include::common/modules/proc_configuring-remaining-smart-proxy-servers-with-defa
...
 include::common/modules/con_configuring-smart-proxy-server-with-custom-ssl-certificates-for-load-balancing-with-puppet.adoc[leveloffset=+2]
 
+:ProductName: {SmartProxyServer}
+:ssl-common-name: {smartproxy-example-com}
+:cert-name: {smart-proxy-context}
 :parent-context: {context}
 :context: load-balancing-with-custom-ssl-certificates-with-puppet
-include::common/modules/proc_creating-custom-ssl-certificates-for-smart-proxy-server.adoc[leveloffset=+3]
+include::common/modules/proc_creating-a-custom-ssl-certificate.adoc[leveloffset=+3]
 :context: {parent-context}
```
Comment (Lennonka) [inline guides/doc-Configuring_Load_Balancer/master.adoc:54]:
```suggestion
:context: {parent-context}
:!parent-context:
```
Let's unset this attribute when it's no longer needed.

**Review by Lennonka — CHANGES_REQUESTED** [review]
It's still missing a ProductName definition somewhere in the docs. Try to look up, where else the module is used and fix it, please.

File: guides/common/assembly_configuring-capsule-custom-server-certificate.adoc [hunk 6]
```diff
@@ -13,8 +13,14 @@ To configure your {SmartProxyServer} with a custom certificate, complete the fol
 . xref:deploying-a-custom-ssl-certificate-to-{smart-proxy-context}-server_{smart-proxy-context}[]
 . xref:deploying-a-custom-ssl-certificate-to-hosts_{smart-proxy-context}[]
 
+:ProductName: {SmartProxyServer}
```
Comment (Lennonka) [inline guides/common/assembly_configuring-capsule-custom-server-certificate.adoc:16]: This is defined at the parent level (in `master.adoc`). So let's remove `:ProductName:` and `:!ProductName:` in this assembly. It should fix the build of the _Installing Proxy_ guide.

File: guides/common/assembly_configuring-satellite-custom-server-certificate.adoc [hunk 7]
```diff
@@ -25,8 +25,14 @@ To configure your {ProjectServer} with a custom certificate, complete the follow
 . If you have external {SmartProxyServers} registered to {ProjectServer}, configure them with custom SSL certificates.
 For more information, see {InstallingSmartProxyDocURL}configuring-capsule-custom-server-certificate_{smart-proxy-context}[Configuring {SmartProxyServer} with a Custom SSL Certificate] in _{InstallingSmartProxyDocTitle}_.
 
+:ProductName: {ProjectServer}
```
Comment (Lennonka) [inline guides/common/assembly_configuring-satellite-custom-server-certificate.adoc:28]: This is defined at the parent level (in `master.adoc`). So let's remove `:ProductName:` and `:!ProductName:` in this assembly. It should fix the build of the _Installing Server_ guide.

**Review by Lennonka — CHANGES_REQUESTED** [review]
Found the problem.

**Review by Lennonka — APPROVED** [review]
LGTM

**Comment by Lennonka** [comment]
Cherry-picked:
- 6549eb96ea..d98121e522  3.7 -> 3.7
- fad69f4f12..83d587cfe1  3.6 -> 3.6
- ad77dca4d3..d11fef0aa8  3.5 -> 3.5

**Pull request merged**